### ISSUE TITLE
Travis not configured properly to parallelize test's 

### ISSUE THREAD

**Issue by kunall17** [zulip/zulip-mobile#1199]
https://docs.travis-ci.com/user/speeding-up-the-build/#Parallelizing-your-builds-across-virtual-machines

**Comment by borisyankov** [comment]
Since setup and booting up the VM for executing tests seems to take the majority of time, I think the current implementation of just parallelizing the runs on a single machine is likely good enough.

**Comment by kunall17** [comment]
The E2E test's are not running currently in the CI

**Comment by borisyankov** [comment]
Yup, but it in CI they were taking more than 6 hours before, consistently.
If we can't make them work within some reasonable time limit, we can't enable them and need to run them manually or something.

**Comment by kunall17** [comment]
6 hours? 
Mostly it took 20-25 mins AFAIR 
https://travis-ci.org/zulip/zulip-mobile/builds/268723075?utm_source=github_status&utm_medium=notification

**Comment by borisyankov** [comment]
It might be the time it takes Travis to start or something. It always took lots and lots of hours between the commit hitting Github and Travis giving it an OK or a Fail.

**Comment by timabbott** [comment]
That sort of outcome is usually because of queuing; if it takes 20-25min for each build, and you're pushing 10/hour (or whatever), it gets backlogged.  They have a fixed number of workers available to each project.

**Comment by borisyankov** [comment]
@kunall17 actually configured a rather special machine they run the tests on, I was surprised Travis even had XCode prepared configurations that can reliably run our E2E tests.

I am not sure if they have very limited number of these machines or if there is some very extended setup process that they undergo, but while our test run times was low, and we weren't pushing many PRs, they still were taking an unpractical amount of time between a PR being made, till it being failed or passed (often 6-12 hours).

**Comment by kunall17** [comment]
This could also be because of the high number of PR's in the GSOC time period & many force pushes, but anyways after better E2E test's I would take this up!

**Comment by borisyankov** [comment]
That was happening consistently even last week before the change.
You can easily test it, by renaming the .travis-e2e.yml to .travis.yml and doing a PR.
